platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dependencies/AbstractExternalModuleDependency.java

```diff
@@ -125,6 +132,25 @@ static ModuleIdentifier assertModuleId(@Nullable String group, @Nullable String
         return DefaultModuleIdentifier.newId(group, name);
     }
 
+    @Override
+    public List<Capability> getRequestedCapabilities() {
+        return getCapabilitySelectors().stream()
+            .map(c -> {
+                if (c instanceof SpecificCapabilitySelector) {
+                    return ((DefaultSpecificCapabilitySelector) c).getBackingCapability();
+                } else if (c instanceof FeatureCapabilitySelector) {
+                    return new DefaultImmutableCapability(
+                        getGroup(),
+                        getName() + "-" + ((FeatureCapabilitySelector) c).getFeatureName(),
+                        getVersion()
+                    );
+                } else {
+                    throw new UnsupportedOperationException("Unsupported capability selector type: " + c.getClass().getName());
+                }
```

Comment:
💭 There's really quite a lot of this selector -> capability conversion going on.  There's got to be a better way.

Reply:
Resolving this comment, as we can discuss as part of a related comment